chia/full_node/mempool_manager.py

```diff
@@ -696,12 +713,23 @@ async def new_peak(
             # find the same transaction multiple times. We put them in a set
             # to deduplicate
             spendbundle_ids_to_remove: Set[bytes32] = set()
+            # Gather also the singleton fast forward puzzle hashes to be removed
+            # from the unspent lineage info map
+            ff_puzzle_hashes_to_remove: Set[bytes32] = set()
             for spend in spent_coins:
                 items: List[MempoolItem] = self.mempool.get_items_by_coin_id(spend)
                 for item in items:
                     included_items.append(MempoolItemInfo(item.cost, item.fee, item.height_added_to_mempool))
                     self.remove_seen(item.name)
                     spendbundle_ids_to_remove.add(item.name)
+                    ff_puzzle_hashes_to_remove = {
```

Comment:
just because one singleton spent made it into the block doesn't mean we don't have more spends of the same singleton left in the mempool. I don't think it's safe to assume we can remove these unless we also make sure there aren't any fast-forward eligible spends left in the mempool.

In fact, it seems like the most important thing we need to do in this case is to *update* the `UnspentLineageInfo`, to still enable us to generate a block with all the ff-eligible coins still in the mempool.

We probably need some cap on how large this can grow, and while producing a block, we nee to tolerate not finding the lineage info we need, and just skip the item.